bftengine/src/bftengine/PersistentStorageImp.hpp

```diff
@@ -94,7 +95,8 @@ const uint16_t numOfLastExitFromViewDescObjs = kWorkWindowSize + 1;
 // objects plus one - for simple descriptor parameters.
 enum DescMetadataParameterIds {
   LAST_EXIT_FROM_VIEW_DESC = WIN_PARAMETERS_NUM + reservedWindowParamsNum,
-  LAST_EXEC_DESC = LAST_EXIT_FROM_VIEW_DESC + numOfLastExitFromViewDescObjs,
+  LAST_COMPLAINTS_DESC = LAST_EXIT_FROM_VIEW_DESC + numOfLastExitFromViewDescObjs,
+  LAST_EXEC_DESC = LAST_COMPLAINTS_DESC + reservedComplaintsNum + 1,
```

Comment:
I don't know if this is important, but LAST_COMPLAINTS_DESC is now the id of the first complaint written to db while previously it was the id of DescriptorOfLastExecution, this seems to break binary compatibility if data written prior to this code update needs to be read.

Reply:
I think the only situation in which this could happen is after update, but currently when we update versions we clear the metadata. Therefore the shifting should be OK, and it preserves the context of the parameter groups.